apps/cowswap-frontend/src/modules/trade/hooks/useSafeBundleFlowContext.ts

```diff
@@ -0,0 +1,47 @@
+import { useMemo } from 'react'
+
+import { getCurrencyAddress } from '@cowprotocol/common-utils'
+import { useSafeAppsSdk } from '@cowprotocol/wallet'
+
+import useSWR from 'swr'
+
+import { useGP2SettlementContract, useTokenContract, useWETHContract } from 'common/hooks/useContract'
+import { useNeedsApproval } from 'common/hooks/useNeedsApproval'
+import { useTradeSpenderAddress } from 'common/hooks/useTradeSpenderAddress'
+
+import { useReceiveAmountInfo } from './useReceiveAmountInfo'
+
+import { SafeBundleFlowContext } from '../types/TradeFlowContext'
+
+export function useSafeBundleFlowContext(): SafeBundleFlowContext | null {
+  const settlementContract = useGP2SettlementContract()
+  const spender = useTradeSpenderAddress()
+
+  const safeAppsSdk = useSafeAppsSdk()
+  const wrappedNativeContract = useWETHContract()
+  const receiveAmountInfo = useReceiveAmountInfo()
+  const inputAmountWithSlippage = receiveAmountInfo?.afterSlippage.sellAmount
+  const needsApproval = useNeedsApproval(inputAmountWithSlippage)
+  const inputCurrencyAddress = useMemo(() => {
+    return inputAmountWithSlippage ? getCurrencyAddress(inputAmountWithSlippage.currency) : undefined
+  }, [inputAmountWithSlippage])
+  const erc20Contract = useTokenContract(inputCurrencyAddress)
+
+  return (
+    useSWR(
```

Comment:
Why `usdSWR` here for what seems to be a regular object?

Reply:
In order to cache the result.
With SWR it doesn't matter how many times you subscribe to the hook, it will always return the same calculation result.